### PR TITLE
Llama 2

### PR DESCRIPTION
**Pull request by AegeanYan**
llama 2 integration
for the `total_len` please check the hf style and i managed to align with it, maybe the llama1 code should be the same with my revision.
the logprob left to be revised soon

### CODE REVIEW

File: examples/rap_blocksworld/inference.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def rap_bw(base_model: LanguageModel,
     import random
     import torch
     import torch.backends.cudnn
-    from reasoners.lm import LLaMAModel
+    from reasoners.lm import LLaMAModel, LlamaModel
```
Comment (Ber666) [inline examples/rap_blocksworld/inference.py:89]: Very confusing

File: reasoners/lm/llama_model_v2.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+        assert bsz <= params.max_batch_size, f"total batch size exceeds limit: {bsz} > {params.max_batch_size}"
+
+        prompt_tokens = [self.tokenizer.encode(x, bos=True, eos=False) for x in inputs]
+
+        min_prompt_size = min([len(t) for t in prompt_tokens])
+        max_prompt_size = max([len(t) for t in prompt_tokens])
+
+        assert max_prompt_size <= params.max_seq_len, f"prompt length exceeds limit: {max_prompt_size} > {params.max_seq_len}"
+        # total_len = min(params.max_seq_len, max_prompt_size + max_new_tokens)
+        # total_len = min(total_len, max_length)##there seems a bug, max_length can be larger than params.max_seq_len. I think we should take min
```
Comment (Ber666) [inline reasoners/lm/llama_model_v2.py:145]: We don't want to show this to users. If you are sure this is a bug please remove the comment.

File: README.md [hunk 3]
```diff
@@ -211,6 +211,7 @@ conda activate reasoners
 
 Clone the repository and install the package:
 
+llama 2 have some alias problem so you may need to clone the forked llama1 at private repo. please check the setup.py
```
Comment (Ber666) [inline README.md:214]: No need to show this in readme. Just put a comment in setup.py is fine.

**Review by Ber666 — COMMENTED** [review]
I made some notes for improvement. Please understand this is a library instead of internal code, i.e. imagine how new users can use the code without any confusions.

File: examples/rap_blocksworld/inference.py (hunk 1 above)
Comment (AegeanYan) [reply examples/rap_blocksworld/inference.py:89]: I just want to align with meta and hf's naming

File: reasoners/lm/llama_model_v2.py (hunk 2 above)
Comment (AegeanYan) [reply reasoners/lm/llama_model_v2.py:145]: ok

**Comment by AegeanYan** [comment]
fixed the mentioned problems while there are some other potential bugs found when I was implementing the log_probs

File: README.md (hunk 3 above)
Comment (AegeanYan) [reply README.md]: I think it may be important to warn our group members or users with older version to re-setup for llama2 integration.

**Comment by AegeanYan** [comment]
check my fix and i think it works

**Pull request merged**